client/blocks/inline-help/inline-help-search-results.jsx

```diff
@@ -15,60 +16,145 @@ import { recordTracksEvent } from 'state/analytics/actions';
 import QueryInlineHelpSearch from 'components/data/query-inline-help-search';
 import PlaceholderLines from './placeholder-lines';
 import { decodeEntities, preventWidows } from 'lib/formatting';
-import getInlineHelpSearchResultsForQuery from 'state/inline-help/selectors/get-inline-help-search-results-for-query';
+import getSearchResultsByQuery from 'state/inline-help/selectors/get-inline-help-search-results-for-query';
 import getSelectedResultIndex from 'state/inline-help/selectors/get-selected-result-index';
+import getInlineHelpCurrentlySelectedResult from 'state/inline-help/selectors/get-inline-help-currently-selected-result';
 import isRequestingInlineHelpSearchResultsForQuery from 'state/inline-help/selectors/is-requesting-inline-help-search-results-for-query';
 import hasInlineHelpAPIResults from 'state/selectors/has-inline-help-api-results';
 import { selectResult } from 'state/inline-help/actions';
 import { localizeUrl } from 'lib/i18n-utils';
+import Gridicon from 'components/gridicon';
+import {
+	SUPPORT_TYPE_ADMIN_SECTION,
+	SUPPORT_TYPE_API_HELP,
+	SUPPORT_TYPE_CONTEXTUAL_HELP,
+} from './constants';
 
 function HelpSearchResults( {
 	hasAPIResults = false,
 	isSearching = false,
-	openResult,
+	onSelect,
 	searchQuery = '',
-	searchResults,
+	searchResults = [],
+	selectedResult = {},
 	selectedResultIndex = -1,
 	selectSearchResult,
 	translate = identity,
 	placeholderLines,
+	track,
 } ) {
-	const selectResultHandler = ( selectionIndex ) => ( event ) => {
-		const selectedResult = searchResults?.[ selectionIndex ] ?? null;
-		selectSearchResult( selectionIndex );
-		openResult( event, selectedResult );
+	const supportTypeRef = useRef( searchResults?.[ 0 ]?.support_type );
+
+	function getTitleBySectionType( addSection, type, query = '' ) {
+		if ( ! addSection ) {
+			return null;
+		}
+
+		let title = '';
+		switch ( type ) {
+			case SUPPORT_TYPE_CONTEXTUAL_HELP:
+				if ( ! query.length ) {
+					return;
+				}
+				title = translate( 'This might interest you' );
+				break;
+
+			case SUPPORT_TYPE_API_HELP:
+				title = translate( 'Support articles' );
+				break;
+			case SUPPORT_TYPE_ADMIN_SECTION:
+				title = translate( 'Show me where to' );
+				break;
+			default:
+				return null;
+		}
+
+		return (
+			<li className="inline-help__results-title">
+				<h2>{ title }</h2>
+			</li>
+		);
+	}
+
+	const onLinkClickHandler = ( event ) => {
+		const { support_type: supportType, link } = selectedResult;
+
+		// check and catch admin section links.
+		if ( supportType === SUPPORT_TYPE_ADMIN_SECTION && link ) {
+			// record track-event.
+			track( 'calypso_inlinehelp_admin_section_search', {
+				link: link,
+				search_term: searchQuery,
+			} );
+
+			// push state only if it's internal link.
+			if ( ! /^http/.test( link ) ) {
+				event.preventDefault();
+				page( link );
+			}
```

Comment:
In `inline-help-search-card.jsx` we also account for external links (see below). Should we do that here too?

```
// push state only if it's internal link.
if ( ! /^http/.test( link ) ) {
    event.preventDefault();
    page( link );
} else {
    // redirect external links.
    window.location.href = link;
}
```

Reply:
No, because it's handled by the client in a regular way, as usual, since the event is not prevented (event.preventDefault()) when it's an external link. Anyway good comment.